[adoubleb] iP

adoubleb opened the pull request:
## BrowniePro 
slayy! :nail_care:

> [!WARNING]
> This is awesome. Says who? [This guy](https://www.fauna-flora.org/species/mountain-gorilla/).

#### What's good
- Equipped with the latest frameworks
* Easy-to-use
+ Functional (mostly)

#### Better than our competitors:
1. Simsima
2. TalkGPT
3. ShallowSeek

#### Clean Code!!

```java
public Brownie() {
        ui = new Ui();
        tasklistManager = new TasklistManager();
        try {
            tasklistManager.initializeTasklist();
            items = tasklistManager.loadTasksFromFile();
        } catch (Exception e) {
            e.printStackTrace();
        }
}
```

_I_ **use** `enums` ~~too~~  :sweat_smile:

#### Weeks
- [x] 1-3
- [ ] 4-13

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

adoubleb committed:
Changed Duke.java to Brownie.java and edited text

adoubleb committed:
added class files to gitignore

adoubleb committed:
Brownie echoes

adoubleb committed:
Brownie saves into a list

adoubleb committed:
Brownie can mark and umark tasks

adoubleb committed:
added different types of tasks

adoubleb committed:
Update Exception handling for invalid commands

Printing error messages manually
Bad practise
Create custom exception for more explicit error messages

adoubleb committed:
Add delete functionality

adoubleb committed:
Add testing

adoubleb committed:
Store tasklist in disk memory

adoubleb committed:
Merge branch 'branch-Level-7'
Merge branch-Level-7 into master

Integrate feature store tasklist into disk memory

adoubleb committed:
Store time and dates as LocalDateTime

adoubleb committed:
Merge branch 'branch-Level-8'
Merge branch branch-Level-8

Time and dates treated as String

Merge will handle them as LocalDateTime

adoubleb committed:
Integrate more OOP

adoubleb committed:
Merge branch 'moreOOP'
Integrate more OOP features

Create Command classes

Create UI class

Add another parsing class

adoubleb committed:
Merge branch 'add-gradle-support'
Add gradle support

adoubleb committed:
Adapt gradle to personal project

adoubleb committed:
Merge branch 'add-gradle-support'

Adapt to personal project

adoubleb committed:
Add junit tests

adoubleb committed:
Merge branch 'add-junit'
Add junit tests

adoubleb committed:
Configure build.gradle to create jar

adoubleb committed:
Add javadoc to Brownie class

adoubleb committed:
Add javadocs to all command classes

adoubleb committed:
Add javadocs to exception class

adoubleb committed:
Add javadocs to TasklistManager

adoubleb committed:
Add javadocs to all parser class

adoubleb committed:
Follow coding standard in brownie class

adoubleb committed:
Follow coding standard in command class

adoubleb committed:
Support find command

adoubleb committed:
Merge branch 'branch-A-javadoc'
Add javadoc to most public classes and methods

adoubleb committed:
Merge branch 'branch-A-CodingStandard'
Change to comply with Java coding standard

adoubleb committed:
Merge branch 'branch-Level-9'
Support find command

adoubleb committed:
Make tasklist manager final

adoubleb committed:
Add and config checkstyle

adoubleb committed:
Merge branch 'branch-A-checkstyle'

Add and config checkstyle

adoubleb committed:
Add gui without integration

adoubleb committed:
Integrate GUI into brownie

adoubleb committed:
Merge branch 'branch-Level-10'
Integrate GUI into brownie

TrKNguyen commented on data/brownie/tasklist.txt:
```diff
@@ -0,0 +1,7 @@
+T | 0 | gym
+D | 0 | iP | 2025-02-02T18:01
+T | 0 | something
```
This storage method makes loading the data a lot easier, nice.

TrKNguyen commented on src/main/java/commands/AddCommand.java:
```diff
@@ -0,0 +1,85 @@
+ */
+public class AddCommand extends Command {
+    private static final String START_DELIMITER = "start:";
+    private static final String BY_DELIMITER = "by:";
```
the way you explicitly declare START_DELIMITER, BY_DELIMITER is good

TrKNguyen commented on src/main/java/commands/AddCommand.java:
Create multiple files for some specific task is very good practice, nice!

TrKNguyen commented on src/main/java/commands/MarkCommand.java:
Instead of return a String, I think if you want to capture the output, you can do like capture the output of System.out.print into a String, you can use ByteArrayOutputStream and PrintStream to redirect System. out

adoubleb replied:
Nice suggestion, thanks!

TrKNguyen commented on src/main/java/iomanager/TasklistManager.java:
```diff
@@ -0,0 +1,179 @@
+        } catch (IOException e) {
+            writer.println("Error while creating tasklist.txt: " + e.getMessage());
+            writer.flush();
+            System.exit(1);
```
I think just throw exception, not System.exit(1),

adoubleb replied:
Good suggestion, thanks!

TrKNguyen commented on src/main/java/iomanager/TasklistManager.java:
```diff
@@ -0,0 +1,179 @@
+    private static final String DATA_FOLDER_PATH = "./data/brownie";
+    private static final String TASKLIST_FILE_PATH = DATA_FOLDER_PATH + "/tasklist.txt";
+
+    private final PrintWriter writer;
```
PrintWriter flushes automatically so I think you dont need to call flush

TrKNguyen commented on src/main/java/iomanager/TasklistManager.java:
```diff
@@ -0,0 +1,179 @@
+     * user interaction and file management tasks.
+     */
+    public TasklistManager() {
+        this.writer = new PrintWriter(System.out);;
```
redundant ;; maybe?

adoubleb replied:
Good catch, thanks!

TrKNguyen commented on src/test/java/parser/ExtractDateTimeTest.java:
Well-structured and good test cases

TrKNguyen reviewed (commented):
Really good overall, some small features can be improved!

CoreySiah commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,22 @@
+package commands;
```
Good work on having separate packages for commands, etc.! 😁

CoreySiah commented on src/main/java/boxes/DialogBox.java:
```diff
@@ -0,0 +1,60 @@
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
```
Good display and organization of imports! 😁

CoreySiah commented on src/main/java/parser/UserInput.java:
```diff
@@ -0,0 +1,109 @@
+        String[] tokens = input.split("\\s+");
+        String commandWord = tokens[0];
+        switch (commandWord) {
+        case "list":
```
Maybe include indentation for the switch cases 😁

CoreySiah commented on src/main/java/task/Event.java:
```diff
@@ -0,0 +1,39 @@
+        this.start = start;
+        this.end = end;
+    }
+    public String mySymbol() {
```
Minor spacing issue between properties, constructor, and methods! 😁

CoreySiah commented on src/main/java/commands/ExitCommand.java:
```diff
@@ -0,0 +1,22 @@
+ * 2. Terminates the application process.
+ * Overrides the execute method from the Command class.
+ */
+public class ExitCommand extends Command {
```
Good use of individual classes for commands! 😁

CoreySiah reviewed (commented):
LGTM! Great work; just minor comments! 😁

adoubleb committed:
Refactor to abide by checkstyle

Imports violate checkstyle order

This violates CS2103 requirements

Let's reorder the imports

adoubleb committed:
Merge branch 'branch-refactoring'

Imports were ordered randomly and random whitespaces existed

This violates the checkstyle by CS2103

Let's change this to follow the checkstyle

adoubleb committed:
Add assertions to check logic

adoubleb committed:
Refactor AddCommand file for readability

adoubleb committed:
Refactor files for readability

adoubleb committed:
Merge pull request #1 from adoubleb/branch-A-Assertions

Add assertions to check logic

adoubleb committed:
Merge master

adoubleb committed:
Merge pull request #2 from adoubleb/branch-A-CodeQuality

Merge branch-A-Code Quality

adoubleb committed:
Add feature detect duplicates

adoubleb committed:
Merge branch 'branch-BCD-Extension'

Add feature to detect duplicates

adoubleb committed:
Add AI.md file

adoubleb committed:
Fix bugs in AddCommand.java file

First word in description was excluded after parsing

Changed parsing to include the first word in the description

adoubleb committed:
Refactor code in test files

adoubleb committed:
Merge branch 'refactoring'

Test files contained badly named boolean variables

Rename boolean variables according to coding quality

adoubleb committed:
Update README.md

adoubleb committed:
Merge pull request #4 from adoubleb/patch-1

Update README.md

adoubleb committed:
Format deadline datetime output

adoubleb committed:
Update docs/readme

adoubleb committed:
Merge branch 'refactoring'

adoubleb committed:
Correct spelling errors

adoubleb committed:
Refactor code to clean-up

adoubleb committed:
Merge branch 'refactoring'

Refactor code and clean-up

adoubleb committed:
Refactor exit command

adoubleb committed:
Remove unused imports

adoubleb committed:
Add AI use declaration

adoubleb committed:
Improve validation for mark/unmark

adoubleb committed:
Improve validation for find

adoubleb committed:
Add integer validation for index

adoubleb committed:
Add clearer error message

adoubleb committed:
Merge branch 'branch-ImproveValidation'

adoubleb committed:
Update README.md
